src/pages/iou/steps/NewRequestAmountPage.tsx

```diff
@@ -88,29 +72,29 @@ function NewRequestAmountPage({route, iou, report, selectedTab}) {
     useEffect(() => {
         if (isEditing) {
             // ID in Onyx could change by initiating a new request in a separate browser tab or completing a request
-            if (prevMoneyRequestID.current !== iou.id) {
+            if (prevMoneyRequestID.current !== iou?.id) {
                 // The ID is cleared on completing a request. In that case, we will do nothing.
-                if (!iou.id) {
+                if (!iou?.id) {
                     return;
                 }
-                Navigation.goBack(ROUTES.MONEY_REQUEST.getRoute(iouType, reportID), true);
+                Navigation.goBack(ROUTES.MONEY_REQUEST_CONFIRMATION.getRoute(iouType, report?.reportID), true);
                 return;
             }
             const moneyRequestID = `${iouType}${reportID}`;
-            const shouldReset = iou.id !== moneyRequestID;
+            const shouldReset = iou?.id !== moneyRequestID;
             if (shouldReset) {
                 IOU.resetMoneyRequestInfo(moneyRequestID);
             }
 
-            if (!isDistanceRequestTab && (_.isEmpty(iou.participants) || iou.amount === 0 || shouldReset)) {
-                Navigation.goBack(ROUTES.MONEY_REQUEST.getRoute(iouType, reportID), true);
+            if (!isDistanceRequestTab && (!iou?.participants?.length || iou?.amount === 0 || shouldReset)) {
+                Navigation.goBack(ROUTES.MONEY_REQUEST_CONFIRMATION.getRoute(iouType, report?.reportID), true);
             }
         }
 
         return () => {
-            prevMoneyRequestID.current = iou.id;
+            prevMoneyRequestID.current = iou?.id;
         };
-    }, [iou.participants, iou.amount, iou.id, isEditing, iouType, reportID, isDistanceRequestTab]);
+    }, [isEditing, iouType, reportID, isDistanceRequestTab, iou, report?.reportID]);
```

Comment:
Why using the whole `iou` object in dependency array instead of only the fields used in the Effect?

Reply:
You're right, I'll apply it right away!